Tests require ImageGradient plugins

I'm running digits-test on the v5.1-dev release (having built Caffe using the NVcaffe instructions), but I get this failure. Can anyone tell me what might cause that?
```
======================================================================
ERROR: test suite for <class 'digits.dataset.generic.test_views.TestImageGradientCreated'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/usr/local/lib/python2.7/dist-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/usr/local/lib/python2.7/dist-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/usr/local/lib/python2.7/dist-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/daryl/DIGITS-5.1-dev/digits/dataset/generic/test_views.py", line 375, in setUpClass
    image_height=cls.IMAGE_HEIGHT)
  File "/home/daryl/DIGITS-5.1-dev/digits/dataset/generic/test_views.py", line 183, in setUpClass
    cls.dataset_id = cls.create_dataset(json=True, **kwargs)
  File "/home/daryl/DIGITS-5.1-dev/digits/dataset/generic/test_views.py", line 89, in create_dataset
    'Dataset creation failed with %s' % rv.status_code)
RuntimeError: Dataset creation failed with 500
-------------------- >> begin captured stdout << ---------------------
WARNING: log_file config option not found - no log file is being saved
2016-11-11 12:01:55 [INFO ] Loaded 0 jobs.

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
engineio: INFO: Server initialized for gevent.
digits: INFO: Loaded 0 jobs.
--------------------- >> end captured logging << ---------------------
```

Thanks @lukeyeager - that resolved the problem. The tests are now failing further on, at digits.model.images.generic.test_views.TestAllInOneNetwork:

ERROR: test suite for <class 'digits.model.images.generic.test_views.TestAllInOneNetwork'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/usr/local/lib/python2.7/dist-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/usr/local/lib/python2.7/dist-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/usr/local/lib/python2.7/dist-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/daryl/DIGITS-5.1-dev/digits/model/images/generic/test_views.py", line 242, in setUpClass
    assert cls.model_wait_completion(cls.model_id) == 'Done', 'create failed'
AssertionError: create failed
-------------------- >> begin captured logging << --------------------
digits: INFO: Job deleted.
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits: INFO: Job deleted.
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
digits.webapp: INFO: Analyze DB (Training Images) task started.
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: INFO: Task subprocess args: "/usr/bin/python2 /home/daryl/DIGITS-5.1-dev/digits/tools/analyze_db.py /tmp/tmpqCDWAP/train_images --force-same-shape"
digits.webapp: INFO: Analyze DB (Training Labels) task started.
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: INFO: Task subprocess args: "/usr/bin/python2 /home/daryl/DIGITS-5.1-dev/digits/tools/analyze_db.py /tmp/tmpqCDWAP/train_labels --force-same-shape"
digits.webapp: INFO: Analyze DB (Validation Images) task started.
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: INFO: Task subprocess args: "/usr/bin/python2 /home/daryl/DIGITS-5.1-dev/digits/tools/analyze_db.py /tmp/tmpqCDWAP/val_images --force-same-shape"
digits.webapp: INFO: Analyze DB (Validation Labels) task started.
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: INFO: Task subprocess args: "/usr/bin/python2 /home/daryl/DIGITS-5.1-dev/digits/tools/analyze_db.py /tmp/tmpqCDWAP/val_labels --force-same-shape"
digits.webapp: WARNING: Analyze DB (Training Labels) unrecognized output: WARNING: log_file config option not found - no log file is being saved
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: DEBUG: Images are 2x1x1
digits.webapp: WARNING: Analyze DB (Training Images) unrecognized output: WARNING: log_file config option not found - no log file is being saved
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: DEBUG: Images are 10x10x1
digits.webapp: INFO: Analyze DB (Training Labels) task completed.
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
digits.webapp: INFO: Analyze DB (Training Images) task completed.
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
digits.webapp: WARNING: Analyze DB (Validation Images) unrecognized output: WARNING: log_file config option not found - no log file is being saved
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: DEBUG: Images are 10x10x1
digits.webapp: WARNING: Analyze DB (Validation Labels) unrecognized output: WARNING: log_file config option not found - no log file is being saved
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: DEBUG: Images are 2x1x1
digits.webapp: INFO: Analyze DB (Validation Images) task completed.
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
digits.webapp: INFO: Analyze DB (Validation Labels) task completed.
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "task update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130500-e13e [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits: INFO: Job complete.
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130503-9977 [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
digits.webapp: DEBUG: Network sanity check - train
digits.webapp: DEBUG: Network sanity check - val
digits.webapp: DEBUG: Network sanity check - deploy
digits.webapp: INFO: Train Caffe Model task started.
socketio: INFO: emitting event "task update" to 20161111-130503-9977 [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130503-9977 [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
digits.webapp: INFO: Task subprocess args: "/home/daryl/caffe/build/tools/caffe train --solver=/tmp/tmpGa1NVm/20161111-130503-9977/solver.prototxt --gpu=0"
socketio: INFO: emitting event "task update" to 20161111-130503-9977 [/jobs]
digits.webapp: ERROR: Train Caffe Model: Check failed: data_
digits.webapp: ERROR: Train Caffe Model: Check failed: data_
digits.webapp: ERROR: Train Caffe Model: Check failed: data_
digits.webapp: ERROR: Train Caffe Model: Check failed: data_
digits.webapp: ERROR: Train Caffe Model task failed with error code -6
socketio: INFO: emitting event "task update" to 20161111-130503-9977 [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130503-9977 [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]
socketio: INFO: emitting event "server update" to job_management [/jobs]
socketio: INFO: emitting event "job update" to 20161111-130503-9977 [/jobs]
socketio: INFO: emitting event "job update" to job_management [/jobs]

--- a/digits/dataset/generic/test_views.py
+++ b/digits/dataset/generic/test_views.py
@@ -4,6 +4,7 @@ from __future__ import absolute_import
 import json
 import os
 import tempfile
+import unittest
 
 from bs4 import BeautifulSoup
 import numpy as np
@@ -179,6 +180,8 @@ class BaseViewsTestWithDataset(BaseViewsTest):
 
     @classmethod
     def setUpClass(cls, **kwargs):
+        if extensions.data.get_extension(cls.EXTENSION_ID) is None:
+            raise unittest.SkipTest('Extension "%s" is not installed' % cls.EXTENSION_ID)
         super(BaseViewsTestWithDataset, cls).setUpClass()
         cls.dataset_id = cls.create_dataset(json=True, **kwargs)
         assert cls.dataset_wait_completion(cls.dataset_id) == 'Done', 'create failed'

